### PR TITLE
fix: Zoom integration doesn't follow the zoom app settings for auto-record

### LINKED ISSUE #15474: [CAL-3943] Zoom integration doesn't follow the zoom app settings for auto-record
<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3943](https://linear.app/calcom/issue/CAL-3943/zoom-integration-doesnt-follow-the-zoom-app-settings-for-auto-record)</sub>

### PR DESCRIPTION
**Pull request by sbsangu**
## What does this PR do?

This PR addresses the issue identified in [CAL-3943] where meetings generated via cal.com to our team's Zoom account do not auto-record despite the Zoom app settings being configured to enable auto-recording. This change ensures that meetings created through cal.com adhere to the Zoom app settings for auto-recording.

- Fixes #15474 (GitHub issue number)
- Fixes CAL-3943 (Linear issue number)

## Mandatory Tasks (DO NOT REMOVE)

- [X] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [x] N/A-I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [x] N/A-I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

## Checklist

- [X] I have read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have checked if my changes generate no new warnings

### CODE REVIEW

File: packages/app-store/zoomvideo/lib/VideoApiAdapter.ts [hunk 1]
```diff
@@ -123,15 +127,17 @@ const ZoomVideoApiAdapter = (credential: CredentialPayload): VideoApiAdapter =>
...
       topic: event.title,
       type: 2, // Means that this is a scheduled meeting
       start_time: dayjs(event.startTime).utc().format(),
       duration: (new Date(event.endTime).getTime() - new Date(event.startTime).getTime()) / 60000,
-      //schedule_for: "string",   TODO: Used when scheduling the meeting for someone else (needed?)
-      timezone: event.organizer.timeZone,
-      //password: "string",       TODO: Should we use a password? Maybe generate a random one?
+     //schedule_for: "string",   TODO: Used when scheduling the meeting for someone else (needed?)
+     timezone: event.organizer.timeZone,
+     //password: "string",       TODO: Should we use a password? Maybe generate a random one?
```
Comment (sean-brydon) [inline packages/app-store/zoomvideo/lib/VideoApiAdapter.ts:140]: I think we can now do something like 

userSettings. schedule_meeting.default_password_for_scheduled_meetings ?? undefined for setting passwords. As this comes back from the api too

File: packages/app-store/zoomvideo/lib/VideoApiAdapter.ts [hunk 2]
```diff
@@ -242,7 +248,7 @@ const ZoomVideoApiAdapter = (credential: CredentialPayload): VideoApiAdapter =>
   return {
     getAvailability: async () => {
       try {
-        // TODO Possibly implement pagination for cases when there are more than 300 meetings already scheduled.
+        // Implement pagination for cases when there are more than 300 meetings already scheduled.
```
Comment (sean-brydon) [inline packages/app-store/zoomvideo/lib/VideoApiAdapter.ts:251]: This can stay as a TODO

**Review by sean-brydon — COMMENTED** [review]
Nice job - lets fix eslint errors + can we try add the default password into the meeting as i think we can do this now as we have the settings for the users new events

**Comment by sbsangu** [comment]
@sean-brydon   does this now resolve the issue?

**Comment by sbsangu** [comment]
@sean-brydon does this now resolve the issue #15474  ?

**Pull request closed without merging**